uint8 ARM_TYPE_GOLD  = 0
uint8 ARM_TYPE_GREEN = 1
string ARM_TYPE_STRINGS = GOLD,GREEN

uint16 JOINT_TYPE_SHOULDER=0
uint16 JOINT_TYPE_ELBOW=1
uint16 JOINT_TYPE_INSERTION=2
uint16 JOINT_TYPE_ROTATION=4
uint16 JOINT_TYPE_PITCH=5
uint16 JOINT_TYPE_GRASP_FINGER1=6
uint16 JOINT_TYPE_GRASP_FINGER2=7
uint16 JOINT_TYPE_YAW=8
uint16 JOINT_TYPE_GRASP=9
string JOINT_TYPE_STRINGS = SHOULDER,ELBOW,INSERTION,ERR_UNUSED,ROTATION,PITCH,FINGER1,FINGER2,YAW,GRASP
uint16 JOINT_TYPE_PITCH_1=10
uint16 JOINT_TYPE_PITCH_2=11
uint16 JOINT_TYPE_PITCH_3=12
uint16 JOINT_TYPE_PITCH_4=13

uint8 TOOL_TYPE_NONE=0
uint8 TOOL_TYPE_GRASPER_10MM=1
uint8 TOOL_TYPE_GRASPER_8MM=2
string TOOL_TYPE_STRINGS = NONE,GRASPER_10MM,GRASPER_8MM

uint8 CONTROLLER_NONE = 0
uint8 CONTROLLER_END_EFFECTOR = 1
uint8 CONTROLLER_CARTESIAN_SPACE = 6

uint8 CONTROLLER_JOINT_POSITION = 5
uint8 CONTROLLER_JOINT_VELOCITY = 2
uint8 CONTROLLER_JOINT_TORQUE = 8

#uint8 CONTROLLER_APPLY_ARBITRARY_TORQUE = 3 #do not use
uint8 CONTROLLER_HOMING_MODE = 4
uint8 CONTROLLER_MULTI_DOF_SINUSOID = 7

string CONTROLLER_STRINGS = NONE,END_EFFECTOR,JOINT_VELOCITY,APPLY_ARBITRARY_TORQUE,HOMING_MODE,JOINT_POSITION,CARTESIAN_SPACE,MULTI_DOF_SINUSOID,JOINT_TORQUE
